include/aspect/simulator.h

```diff
@@ -647,8 +648,10 @@ namespace aspect
        * @param advection_field Determines whether this variable should select
        * the temperature field or a compositional field.
        */
-      void get_artificial_viscosity (Vector<float> &viscosity_per_cell,
-                                     const AdvectionField &advection_field) const;
+      template <typename T>
+      void get_artificial_viscosity (Vector<T> &viscosity_per_cell,
+                                     const AdvectionField &advection_field,
+                                     const bool use_smoothing = false) const;
```

Comment:
Yes. Just remove the argument and let the function take use the parameter as stored in the `use_smoothing` variable. I don't think this function should change its signature at all.